admin_guide/advanced_scheduling/node_affinity.adoc

```diff
@@ -0,0 +1,153 @@
+
+Node affinity allows you to constrain which nodes your pod is eligible to be scheduled on using labels on the node. 
+
+There are currently two types of node affinity rules, _required_ and _preferred_. 
+
+Required rules *must* be met before a pod can be scheduled on a node. Preferred rules specify that, if the rule is met, the scheduler will try to enforce the rules, but will not guarantee enforcement.
+
+[[admin-guide-sched-affinity-config]]
+== Configuring Node Infinity
+
+Node affinity is configured through the pod specification file. You can specify a xref:admin-guide-sched-affinity-config-req[required rule], a xref:admin-guide-sched-affinity-config-pref[preferred rule], or both. If you specify both, the node must first meet the required rule, then meet the preferred.
+
+The following example is a node specification with a required rule that the pod can only be placed on a node with a label whose key is `kubernetes.io/e2e-az-name` and whose value is either `e2e-az1` or `e2e-az2`. The specification also contains a preferred rule that a node with a label whose key is `another-node-label-key` and whose value is either `another-node-label-value` is preferred for the pod.
+
+.Example pod config file with node affinity
+====
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: with-node-affinity
+spec:
+  affinity: 
+    nodeAffinity: <1>
+      requiredDuringSchedulingIgnoredDuringExecution: <2>
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: kubernetes.io/e2e-az-name <3>
+            operator: In <4>
+            values: <5>
+            - e2e-az1
+            - e2e-az2
+      preferredDuringSchedulingIgnoredDuringExecution: <6>
+      - weight: 1
+        preference:
+          matchExpressions:
+          - key: another-node-label-key <3>
+            operator: In <4>
+            values: <5>
+            - another-node-label-value
+  containers:
+  - name: with-node-affinity
+    image: gcr.io/google_containers/pause:2.0
+----
+====
+<1> The stanza to configure node affility.
+<2> This parameter creates a required rule.
+<3> The key that must be matched to apply the rule. 
+<4> The xref:../../rest_api/openshift_v1.html#unversioned-labelselectorrequirement[operator] represents the relationship between the key and a set of values.  Can be `In`, `NotIn`, `Exists`, or `DoesNotExist`.
+<5> The value that must be matched to apply the rule.
+<6> This parameter creates a preferred rule using the same parameters as a required rule. 
+
+[NOTE]
+====
+If labels on a node change at runtime that results in an node affinity rule on a pod no longer being met, the pod will still continue to run on the node. 
```

Comment:
Would be good to explicit state that it is true for requiredDuringSchedulingIgnoredDuringExecution and preferredDuringSchedulingIgnoredDuringExecution. Other variants *requiredDuringExecution is not yet implemented.